[BUG] Reddit API gives 403 error in local, while generating access token from refresh token.

**Describe the bug**
Reddit API gives **403 error** with description saying **"you've been blocked by network security..."** while generating access token from refresh token.

**To Reproduce**
Steps to reproduce the behavior:

1. Run the Reddit workflow in local docker machine, **after a day**.
2. See error

**Expected behavior**
##### Console logs:
> Failed to obtain credential for tool reddit_get_posts: Failed to fetch access token: 401 {"error":"Failed to refresh access token"}

##### Docker Logs:

> [2025-11-06T06:00:01.470Z] [ERROR] [OAuth] Token refresh failed: {"status":403,"error":"...."}
> [2025-11-06T06:00:01.473Z] [ERROR] [OAuth] Error refreshing token: {"error":{}}
> 
> [2025-11-06T06:00:01.473Z] [ERROR] [OAuthUtilsAPI] [e6825f39] Failed to refresh token for credential
> 
> [2025-11-06T06:00:01.476Z] [ERROR] [OAuthUtilsAPI] [e6825f39] Refresh failed and no valid token found in DB {"message":"Failed to refresh token"}
> 
> [2025-11-06T06:00:01.477Z] [ERROR] [OAuthTokenAPI] [e6825f39] Failed to refresh access token: {"message":"Failed to refresh token"}

**Probable Solution**
For some users passing "**User-Agent**" in header worked for them.

References:
- https://github.com/ReVanced/revanced-patches/issues/4549
- https://stackoverflow.com/a/68598054

I want to try this but my Mac Mini is not capable to run and build 'docker-compose.local.yml'. It always shows error saying "Process killed because of low memory or some similiar error"

--- a/apps/sim/lib/oauth/oauth.test.ts
+++ b/apps/sim/lib/oauth/oauth.test.ts
@@ -214,6 +214,17 @@ describe('OAuth Token Refresh', () => {
       const [, requestOptions] = (mockFetch as Mock).mock.calls[0]
       expect(requestOptions.headers.Accept).toBe('application/json')
     })
+
+    it('should include User-Agent header for Reddit requests', async () => {
+      const refreshToken = 'test_refresh_token'
+
+      await refreshOAuthToken('reddit', refreshToken)
+
+      const [, requestOptions] = (mockFetch as Mock).mock.calls[0]
+      expect(requestOptions.headers['User-Agent']).toBe(
+        'sim-studio/1.0 (https://github.com/simstudioai/sim)'
+      )
+    })
   })
 
   describe('Error Handling', () => {

--- a/apps/sim/lib/oauth/oauth.ts
+++ b/apps/sim/lib/oauth/oauth.ts
@@ -1327,6 +1327,9 @@ function getProviderAuthConfig(provider: string): ProviderAuthConfig {
         clientId,
         clientSecret,
         useBasicAuth: true,
+        additionalHeaders: {
+          'User-Agent': 'sim-studio/1.0 (https://github.com/simstudioai/sim)',
+        },
       }
     }
     case 'wealthbox': {

